vars/cfManifestSubstituteVariables.groovy

```diff
@@ -0,0 +1,125 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.variablesubstitution.ExecutionContext
+import com.sap.piper.variablesubstitution.DebugHelper
+import com.sap.piper.variablesubstitution.YamlUtils
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS + [
+    /**
+     * The `String` path of the Yaml file to replace variables in.
+     * Defaults to "manifest.yml" if not specified otherwise.
+     */
+    'manifestFile',
+    /**
+     * The `String` path of the Yaml file to produce as output.
+     * If not specified this will default to `manifestFile` and overwrite it.
+     */
+    'outputManifestFile',
+    /**
+     * The `String` path of the Yaml file containing the variables' values to use as a replacement in the manifest file.
+     * Defaults to `manifest-variables.yml` if not specified otherwise.
+     */
+    'variablesFile'
+]
+
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * Step that substitutes variables in a given YAML file with those specified in a another. The format to reference a variable
+ * in the YAML file is to use double parentheses `((` and `))`, e.g. `((variableName))`.
+ * A declaration of a variable and assignment of its value is simply done as a property in the variables YAML file.
+ *
+ * The format follows <a href="https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#variable-substitution">Cloud Foundry standards</a>.
+ *
+ * The step is activated by the presence of both a `manifest.yml` and a variables file. Names of both files are configurable.
+ *
+ * Usage: `cfManifestSubstituteVariables manifestFile: "path/to/manifest.yml", variablesFile:"path/to/manifest-variables.yml", script: this`
```

Comment:
Usage is described below under `Examples` (that is a common pattern for all the steps.

Reply:
Ok, I will remove usage here then. Or maybe even more. See comment above.